lib/axon/loop.ex

```diff
@@ -1069,6 +1064,8 @@ defmodule Axon.Loop do
               {:halt, {:halt_loop, state}}
 
             {:continue, state} ->
+              state = %{state | iteration: iters + 1}
```

Comment:
This change is so we always log the first iteration (aka iteration 0). Otherwise the user may not have any feedback for a while.

Reply:
It also makes it consistent with epoch, where the epoch completed is called with epoch 0 and not the next epoch.